testsuite/manualmode/src/test/java/org/jboss/as/test/manualmode/logging/AbstractLoggingTestCase.java

```diff
@@ -114,7 +126,18 @@ static JavaArchive createDeployment(final Class<? extends ServiceActivator> serv
             manifest.append("Dependencies: io.undertow.core");
         }
         archive.addAsResource(new StringAsset(manifest.toString()), "META-INF/MANIFEST.MF");
-        return archive;
+        return addPermissions(archive);
+    }
+
+    public static JavaArchive addPermissions(final JavaArchive archive, final Permission... additionalPermissions) {
+        final Permission[] permissions;
+        if (additionalPermissions == null || additionalPermissions.length == 0) {
+            permissions = DEFAULT_PERMISSIONS;
+        } else {
+            permissions = Arrays.copyOf(DEFAULT_PERMISSIONS, DEFAULT_PERMISSIONS.length + additionalPermissions.length);
+            System.arraycopy(additionalPermissions, 0, permissions, DEFAULT_PERMISSIONS.length, additionalPermissions.length);
+        }
+        return archive.addAsManifestResource(PermissionUtils.createPermissionsXmlAsset(permissions), "permissions.xml");
```

Comment:
The DEFAULT_PERMISSIONS are all for what UndertowServiceActivator does, right? If so, how about moving this to UndertowServiceActivator? ModuleResourceRootPathsTestCase can then call it too, plus another tests we add that use the UndertowServiceActivator facility.